pkg/monitor/worker.go

```diff
@@ -18,6 +20,30 @@ import (
 	"github.com/Azure/ARO-RP/pkg/util/restconfig"
 )
 
+// This function will continue to run until such time as it has a config to add to the global Hive shard map
+// Note that because the mon.hiveShardConfigs[shard] is set to `nil` when its created, the cluster
+// monitors will simply ignore Hive stats until this function populates the config
+func (mon *monitor) populateHiveShardRestConfig(ctx context.Context, shard int) {
+	var hiveRestConfig *rest.Config
+	var err error
+
+	for {
+		hiveRestConfig, err = mon.liveConfig.HiveRestConfig(ctx, shard)
+		if hiveRestConfig != nil {
+			mon.setHiveShardConfig(shard, hiveRestConfig)
+			return
+		}
+
+		mon.baseLog.Warn(fmt.Sprintf("error fetching Hive kubeconfig for shard %d", shard))
```

Comment:
This will be a nice log to end up in a dashboard, I propose we combine the two logs (this warn and the below error). By having static text along with the the error returned it will make identifying what the error was when fetching the kubeconfig easier.

Reply:
So the reason for keeping them separate is for generating a Sev 1 alert when we get N of these consecutively, which is why I intentionally kept the low level alert out of it. Isn't the clean separation better?

Reply:
I'm ok with that reasoning, thank you for the clarification.